### PR TITLE
[vcpkg script] Test pkg-config integration

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Test user-perceived `pkg-config` integration in vcpkg (`find_package(PkgConfig)`, `pkg_check_modules(...)`).

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  not needed.

#### Related

https://github.com/microsoft/vcpkg/pull/22392#issuecomment-1162724678 
https://github.com/microsoft/vcpkg/pull/25529

### CODE REVIEW

**Comment by dg0yt** [comment]
Host `pkgconf` can be found reliably with CMake 3.22 `FindPkgConfig.cmake` (but only after system `pkg-config`). But there is room for problems in user projects:
- Our tests do set `VCPKG_HOST_TRIPLET`. Users might miss this parameter, so that `CMAKE_PROGRAM_PATH` wouldn't be set to the host `tools`. Solution for manifest mode: #25529.
- User's CMake before 3.22 doesn't look for `pkgconf`. Possible solutions:
  - Rename `pkgconf[.exe]` to `pkg-config[.exe]`. This would bring vcpkg's port in front of system tools, for all triplets, increasing decoupling from system libraries.
  - Add a `pkgconfig.bat` wrapper which forwards to `pkgconf`, for Windows.

**Comment by Neumann-A** [comment]
> User's CMake before 3.22

Don't need to care about it. System pkg-config is ok on !Windows. Windows users should update cmake any way. 

Main problem with current FindPkgConfig in vcpkg is that it doesn't support multi-config generators yet. We could do that with a wrapper installed alongside with pkgconf which does overrides the functions cmake normally uses and setup the targets correctlyy dependening on config.

**Comment by dg0yt** [comment]
> > User's CMake before 3.22
> 
> Don't need to care about it. System pkg-config is ok on !Windows. Windows users should update cmake any way.

I mostly agree, but I don't want to decide for all users.
A possible issue with system pkg-config on !windows is that it can hide problems with vcpkg pc modules.

> Main problem with current FindPkgConfig in vcpkg is that it doesn't support multi-config generators yet. We could do that with a wrapper installed alongside with pkgconf which does overrides the functions cmake normally uses and setup the targets correctlyy dependening on config.

The problem with wrappers vs. port `pkgconf` is that wrappers are loaded for the target triplet, but `pkgconf` is used as a host dependency.
More generally, the problem is that you cannot know how the results of `pkg_check_modules` are used. As with CMake find modules (and wrappers), it is probably easy to carry multi-config with imported targets, but it is error-prone to handle multi-config for the variables. A modified behaviour wouldn't only affect the current project, but also its transitive dependencies.

(I have a local branch which additonally uses the Ninja multi-config generator for this test-port, but I didn't see a benefit for `find_package` tests.)

**Review by LilyWangLL — DISMISSED** [review]

File: scripts/test_ports/cmake-user/project/FindZLIBviaPkgConfig.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# `pkgconf` is not recognized before CMake 3.22
+find_program(PKG_CONFIG_EXECUTABLE NAMES pkgconf)
```
Comment (JackBoosY) [inline scripts/test_ports/cmake-user/project/FindZLIBviaPkgConfig.cmake:2]: Should we use `REQUIRED` here?

Comment (dg0yt) [reply scripts/test_ports/cmake-user/project/FindZLIBviaPkgConfig.cmake:2]: Makes sense, given the dependency on port `pkgconf`.

Comment (dg0yt) [reply scripts/test_ports/cmake-user/project/FindZLIBviaPkgConfig.cmake]: Done.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
I think vcpkg has the ability to provide usage information for pkg-config by now.
Next we should add usage printing code about pkg-config to vcpkg-tool.

**Pull request merged**